src/Campaigns/resources/admin/components/CampaignDetailsPage/index.tsx

```diff
@@ -132,6 +135,26 @@ export default function CampaignsDetailsPage({campaignId}) {
         });
     }, [campaign?.status]);
 
+    // Show campaign page draft notice
+    useEffect(() => {
+        // @ts-ignore
+        if (record?.status === 'publish') {
+            return;
+        }
+
+        // @ts-ignore
+        if (record && record?.status !== 'publish' && campaign?.status !== 'archived') {
+            dispatch.addNotice({
+                id: 'update-campaign-draft-page-notice',
+                type: 'info',
+                isDismissible: false,
+                content: <DraftCampaignPageNotice />,
+            });
+        } else {
+            dispatch.dismissNotification('update-campaign-draft-page-notice');
+        }
+    }, [record, campaign?.status]);
```

Comment:
When adding deps to useEffect we want to avoid adding objects as they will always be falsy and cause re-renders so it's best to specify the specific property/value you are watching.

```suggestion
    }, [record?.status, campaign?.status]);
```